### PR TITLE
Implement finalize for WandbLogger

### LINKED ISSUE #3002: Checkpoint files location when WandbLogger
I am trying to look for checkpoint files when using WandbLogger. However checkpoint files doesn't seem to be exist in wandb folder as well. This is despite the fact that log_model=True is setup in WandbLogger. Please help me on this, I am not sure if I am missing something on this.

OS is macOS Mojave. However I noticed the issue on Colab, didn't run on my local machine though.

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #3002

The checkpoint artifacts will be uploaded at the end of the run:
image

### CODE REVIEW

File: pytorch_lightning/loggers/wandb.py [hunk 1]
```diff
@@ -156,3 +156,7 @@ def name(self) -> Optional[str]:
     def version(self) -> Optional[str]:
         # don't create an experiment if we don't have one
         return self._experiment.id if self._experiment else self._id
+
+    def finalize(self, status: str) -> None:
+        if self._log_model:
```
Comment (Borda) [inline pytorch_lightning/loggers/wandb.py:161]:
```suggestion
        # upload all checkpoints from saving dir
        if self._log_model:
```

**Review by Borda — APPROVED** [review]

**Review by ydcjeff — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Comment by borisdayma** [comment]
Thanks @awaelchli 

I see from the commit history you probably intended to call `self.experiment.logger.finish()`.

The reason I didn't add it in the logger is I was scared it would affect users who manually upload other charts (interpretation, etc) or files to their W&B run once training is complete.

I think we will be able to make the model upload cleaner through W&B artifacts.
I plan to look at it and propose something in the coming weeks.

**Comment by awaelchli** [comment]
@borisdayma Great to have you look into it more!
I tried `self.experiment.finish()` but this run object didn't have the finish method. The reason I ended up with wandb.save is because I saw it in the docs and it seemed to be a good way. For now, it is guarded by `self._log_model` so people can keep it off if there is an issue with it and we can look into your proposal :)

**Comment by borisdayma** [comment]
@awaelchli Thanks for the feedback. It would be easy add `finish` to `self.experiment` but I don't think it's a good idea.
For now let's keep your implementation and we'll try to find a better way with artifacts.
Thanks a lot for the help!

**Pull request merged**